File: C++/find-the-student-that-will-replace-the-chalk.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        k %= accumulate(cbegin(chalk), cend(chalk), 0LL);
        for (int i = 0; i < size(chalk); k -= chalk[i++]) {
            if (k < chalk[i]) {
                return i;
            }
        }
        return -1;
    }
};

File: Python/find-the-student-that-will-replace-the-chalk.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def chalkReplacer(self, chalk, k):
        """
        :type chalk: List[int]
        :type k: int
        :rtype: int
        """
        k %= sum(chalk)
        for i, x in enumerate(chalk):
            if k < x:
                return i
            k -= x
        return -1
